pkg/spdk/replica.go

```diff
@@ -249,6 +252,17 @@ func (r *Replica) Sync(spdkClient *spdkclient.Client) (err error) {
 			if bdevLvol.DriverSpecific.Lvol.Xattrs[spdkclient.SnapshotChecksum] != "" {
 				continue
 			}
+			parentBdevLvol := bdevLvolMap[bdevLvol.DriverSpecific.Lvol.BaseSnapshot]
+			if bdevLvol.DriverSpecific.Lvol.Xattrs[spdkclient.UserCreated] == "false" || (parentBdevLvol != nil && parentBdevLvol.DriverSpecific.Lvol.Xattrs[spdkclient.UserCreated] == "false") {
+				// Skip the checksum calculation of system created snapshot lvols during rebuilding as they may be purged later.
+				if r.isRebuilding {
```

Comment:
I am a little bit confused: do we skip or do we delay the checksum calculation of system created snapshot?

Reply:
At the end of each rebuilding, Longhorn will issue a snapshot purge request, which will try its best to remove all system snapshots. In other words, calculating these system snapshot checksums does not make sense as they will be removed soon. This delay would make the system snapshot deletion be handled before the calculation starts.